### ISSUE TITLE
Nokee plugin should support headers-only libraries

### ISSUE THREAD

**Issue by max-ag** [nokeedev/gradle-native#105]
A lot of libraries out there are headers only, which isn't currently supported by the Nokee plugins.
The current workaround is to add a dummy .c/.cpp file to the library, but this isn't ideal.
Headers-only libs shouldn't need to be compiled, as there isn't anything to compile.

**Comment by lacasseio** [comment]
I totally agree this is a completely counter-productive workaround. The current leading idea is to use the component API which will offer flexibility in declaring components to be built. In that case the linkage would be set to none meaning no binaries should be produced.

A header only library could be provided unfortunately the biggest open question resolve around C++ vs C header and how to handle the difference. Having two plugins for each type of header would be a bit confusing to the users. We definitely need to introduce a plugin for the common use case but we need to figure out the intension from the developers perspective.
